app/services/adviser/sign_up.rb

```diff
@@ -0,0 +1,49 @@
+class Adviser::SignUp
+  class AdviserSignUpUnavailableError < RuntimeError; end
```

Comment:
Swap out `RuntimeError` for `StandardError`.

Reply:
What's the reasoning for this? In my mind this error should only happen if an application form has been in a valid state and unexpectedly transitions to an invalid state (for an adviser) prior to/as the sign up happens (after the latest eligibility check, which would be on the POST action of the controller), which is a runtime error?